Add a Podspec for GTM.

perotinus opened the pull request:
This only adds a Podspec for some of the commonly-used pieces of GTM.

Currently, there are some issues with some of the subspecs. These subspecs are either commented-out or annotated with a TODO. The rest of the subspecs should be fine.

I tested this locally by referencing

There is currently no 2.0.0 tag. We can either change the version here, decide to cut a 2.0.0 release arbitrarily, or perhaps cut a beta release tag for testing and then do an actual 2.0.0 release.

perotinus commented:
I'm a Googler. I've added my email address to me account, and this commit should have used my corporate email.

thomasvl commented:
btw - when we get this sorted out, can you squash these into a single commit to land?  (git rebase -i HEAD~# (# is 4 at the moment))

perotinus commented:
OK, I've updated this PR with:
- reordered subspecs
- more dependencies extracted
- tighter dependency structure
- a different tag format

thomasvl commented:
@perotinus I think there's a few replies still in your court.  (and the squash request as the commits are building up making it harder to look at what the real changes are)

perotinus commented:
OK, I've squashed everything into one commit, and responded to the inline comments.

perotinus commented:
Sorry, I had to rebase! This should be up-to-date, and include all fixes from the comments.

* I removed the TODO from the KVO code; your suggestion that people who care can edit the source code sounds reasonable to me
* I've replaced the commented-out rules for SQLite and GTMLogger+ASL with comments
